### PR TITLE
Removing Util.dart and moving its remaining parts to better places

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Fixes #623 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on the latest `master`
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: doc/util.md [hunk 1]
```diff
@@ -1,37 +1,39 @@
 # Util
 
-Some stuff just doesn't fit anywhere else.
+On this page you can find documentation for some deviceities classes and methods.
```
Comment (wolfenrain) [inline doc/util.md:3]: `deviceities` is not a word in my book ;-). I suspect you meant something about devices?

File: doc/util.md [hunk 2]
```diff
@@ -1,37 +1,39 @@
...
-It is recommended that the functions in this class be called via the `Flame.util` getter to utilize a single instance prepared by the Flame engine.
-
-### `Flame.util.fullScreen()`
+### `Flame.device.fullScreen()`
 
 When called, this disables all `SystemUiOverlay` making the app full screen.
 When called in the main method, makes your app full screen (no top nor bottom bars)
 
-### `Flame.util.setLandscape()`
+_Has no effect when called on web._
```
Comment (wolfenrain) [inline doc/util.md:14]: I would sugest to make this into a note as we have done in other places:
```markdown
**Note:** Has no effect when called on the web.
```

File: doc/util.md [hunk 3]
```diff
@@ -1,37 +1,39 @@
...
+
+If a finer control of the allowed orientations is required (without having to deal with `SystemChrome` directly), `setOrientation` (accepts a single `DeviceOrientation` as a parameter) and `setOrientations` (accepts a `List<DeviceOrientation>` for possible orientations) can be used.
 
-* `renderWhereAt` and `renderRotated`: if you are directly rendering to the `Canvas`, you can use these functions to easily manipulate coordinates to render things on the correct places. They change the `Canvas` transformation matrix but reset afterwards.
+_Has no effect when called on web._
  
 ## Timer
 
-Flame provides a simple utility class to help you handle countdowns and event like events.
+Flame provides a simple deviceity class to help you handle countdowns and event like events.
```
Comment (wolfenrain) [inline doc/util.md:36]: :-) `device`

File: doc/util.md [hunk 4]
```diff
@@ -136,7 +138,7 @@ class MyBaseGame extends BaseGame {
 
 ## Extensions
 
-Flame bundles a collection of utility extensions, these extensions are meant to help the developer with shortcuts and converion methods, here you can find the summary of those extensions
+Flame bundles a collection of deviceity extensions, these extensions are meant to help the developer with shortcuts and converion methods, here you can find the summary of those extensions
```
Comment (wolfenrain) [inline doc/util.md:141]: `device`

File: lib/src/device.dart [hunk 5]
```diff
@@ -1,20 +1,17 @@
 import 'dart:async';
-import 'dart:ui';
-
 import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
 
-import 'extensions/vector2.dart';
-
-/// Some utilities that did not fit anywhere else.
+/// Class with utilities methods for controlling the device somehow (e.g. setting the screen as fullscreen)
```
Comment (wolfenrain) [inline lib/src/device.dart:5]: Provides methods for controlling the device (e.g. setting the screen to fullscreen).

**Review by spydon — APPROVED** [review]
Nice cleanup!

**Review by wolfenrain — APPROVED** [review]
LGTM, few doc nits.

File: doc/util.md (hunk 1 above)
Comment (erickzanardo) [reply doc/util.md:3]: Lol, I think I mixed utilities and device, thanks for spotting that

Comment (erickzanardo) [reply doc/util.md:3]: Oh, I see what happened, I was quite naive on trusting that a replace all would fix everything quicker! hahaha

File: doc/util.md [hunk 6]
```diff
@@ -145,3 +147,4 @@ They can all be imported on `package:flame/extensions/...`
 Methods:
  - `scaleVector`: Just like `canvas scale` method, but takes a `Vector2` as an argument.
  - `translateVector`: Just like `canvas translate` method, but takes a `Vector2` as an argument.
+ - `rendereAt` and `renderRotated`: if you are directly rendering to the `Canvas`, you can use these functions to easily manipulate coordinates to render things on the correct places. They change the `Canvas` transformation matrix but reset afterwards.
```
Comment (wolfenrain) [inline doc/util.md:150]: `renderAt`

File: lib/src/device.dart [hunk 7]
```diff
@@ -1,20 +1,17 @@
...
-/// Some utilities that did not fit anywhere else.
+/// Provides methods for controlling the device (e.g. setting the screen to fullscreen).
 ///
-/// To use this class, access it via [Flame.util].
-class Util {
+/// To use, access it via [Flame.device]
+class Device {
   /// Sets the app to be fullscreen (no buttons bar os notifications on top).
   ///
   /// Most games should probably be this way.
```
Comment (wolfenrain) [inline lib/src/device.dart:11]: I know this comment was already existing. But I think we can just remove this. It is up to the developer to decide that. This feels a bit judging.. "If you dont do this, you are doing it wrong". And it does not really provide any useful information about the method itself.

Comment (erickzanardo) [reply lib/src/device.dart]: Agree! Removed!

**Review by wolfenrain — APPROVED** [review]
LGTM

**Review by renancaraujo — APPROVED** [review]

File: doc/examples/debug/lib/main.dart [hunk 8]
```diff
@@ -1,13 +1,12 @@
...
 import 'package:flame/game.dart';
-import 'package:flame/flame.dart';
 
 import 'package:flutter/material.dart' hide Image;
 
 import 'dart:ui';
 
 void main() async {
-  Flame.initializeWidget();
+  WidgetsFlutterBinding.ensureInitialized();
```
Comment (luanpotter) [inline doc/examples/debug/lib/main.dart:9]: I thought we didn't need this anymore

File: lib/src/extensions/canvas.dart [hunk 9]
```diff
@@ -12,4 +12,33 @@ extension CanvasExtension on Canvas {
...
   }
+
+  /// Utility method to render stuff on a specific place in an isolated way.
+  ///
+  /// Some render methods don't allow to pass a vector.
+  /// This method translate the canvas before rendering your fn.
+  /// The changes are reset after the fn is run.
+  void renderAt(Vector2 p, void Function(Canvas) fn) {
+    save();
+    translate(p.x, p.y);
```
Comment (luanpotter) [inline lib/src/extensions/canvas.dart:23]: nit: translateVector

File: lib/src/extensions/canvas.dart [hunk 10]
```diff
@@ -12,4 +12,33 @@ extension CanvasExtension on Canvas {
...
+  ///
+  /// It rotates the canvas around the center of rotation.
+  /// The changes are reset after the fn is run.
+  void renderRotated(
+    double angle,
+    Vector2 rotationCenter,
+    void Function(Canvas) fn,
+  ) {
+    save();
+    translate(-rotationCenter.x, -rotationCenter.y);
```
Comment (luanpotter) [inline lib/src/extensions/canvas.dart:38]: ditto

File: lib/src/device.dart [hunk 11]
```diff
@@ -1,20 +1,15 @@
...
-/// To use this class, access it via [Flame.util].
-class Util {
+/// To use, access it via [Flame.device]
+class Device {
   /// Sets the app to be fullscreen (no buttons bar os notifications on top).
-  ///
-  /// Most games should probably be this way.
   Future<void> fullScreen() {
     if (kIsWeb) {
+      // TODO: We probably could use dart:html and implement this for web as well
```
Comment (luanpotter) [inline lib/src/device.dart:12]: can JS change the browser window size? I wonder if this TODO still makes sense

File: doc/examples/debug/lib/main.dart (hunk 8 above)
Comment (erickzanardo) [reply doc/examples/debug/lib/main.dart:9]: We don't need initialDimensions, this is still needed when we load assets before the runApp

File: lib/src/device.dart (hunk 11 above)
Comment (erickzanardo) [reply lib/src/device.dart:12]: Make the browser fullscreen != change its window dimensions.

I have already used JS to make the browser fullscreen in plain JS, so this TODO do make sense. I plan on do a follow up to add this feature after this is merged.

File: doc/util.md [hunk 12]
```diff
@@ -1,34 +1,36 @@
 # Util
 
-Some stuff just doesn't fit anywhere else.
+On this page you can find documentation for some utilities classes and methods.
```
Comment (spydon) [inline doc/util.md:3]: utility classes*

File: lib/src/device.dart [hunk 13]
```diff
@@ -1,20 +1,15 @@
...
 import 'package:flutter/services.dart';
 
-import 'extensions/vector2.dart';
-
-/// Some utilities that did not fit anywhere else.
+/// Provides methods for controlling the device (e.g. setting the screen to fullscreen).
 ///
-/// To use this class, access it via [Flame.util].
-class Util {
+/// To use, access it via [Flame.device]
```
Comment (spydon) [inline lib/src/device.dart:7]: I think it was better before with "To use this class"

File: lib/src/device.dart [hunk 14]
```diff
@@ -1,20 +1,15 @@
...
+class Device {
   /// Sets the app to be fullscreen (no buttons bar os notifications on top).
```
Comment (spydon) [inline lib/src/device.dart:9]: Not your change, but within the parenthesis it should be something like:
"no buttons, bar or notifications on top"

**Review by spydon — APPROVED** [review]
Found some minor grammar stuff, otherwise it looks good!

**Pull request merged**